mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,449 @@
+
+    @classmethod
+    def post_run(cls, context: MLClientCtx):
+        """
+        Performs post-run tasks for logging MLFlow artifacts generated during the run.
+
+        :param context: Current mlrun context
+        """
+        # Get the last run (None means no run was created or found):
+        run = mlflow.last_active_run()
+        if run is None:
+            return
+
+        # Log the run:
+        MLFlowTracker._log_run(context=context, run=run)
+
+    @classmethod
+    def import_run(
+        cls,
+        project: MlrunProject,
+        pointer: str,
+        function_name: str = None,
+        handler: str = None,
+    ) -> RunObject:
+        """
+        Import a previous MLFlow experiment run to MLRun.
+
+        :param project:       The MLRun project to import the run to.
+        :param pointer:       The MLFlow `run_id` to import.
+        :param function_name: The MLRun function to assign this run to.
+        :param handler:       The handler for MLRun's RunObject
+
+        :return: The newly imported RunObject.
+        """
+        # Validate function name was given:
+        if function_name is None:
+            raise ValueError(
+                "For importing a MLFlow experiment run, a MLRun function name must be provided."
+            )
+
+        # Get the MLFlow run object:
+        run = mlflow.search_runs(
+            search_all_experiments=True,
+            filter_string=f"attributes.run_id = '{pointer}'",
+            output_format="list",
+        )
+        if not run:
+            raise ValueError(
+                "The provided run id was not found in the set MLFlow registry. Try to set the registry using "
+                "MLFlow's environment variables."
+            )
+        run = run[0]  # We are using a run id, so only one will be returned in the list.
+        # Create a run spec and metadata for creating the run object to hold the mlflow run:
+        run_spec = RunSpec(function=function_name, handler=handler)
+        run_metadata = RunMetadata(
+            uid=run.info.run_uuid, name=run.info.run_name, project=project.name
+        )
+        run_object = RunObject(spec=run_spec, metadata=run_metadata)
+        # Create a context from the run object:
+        ctx = mlrun.get_or_create_ctx(
+            name=run.info.run_name,
+            spec=run_object,
+        )
+        # Store the run in the MLRun DB, then import the MLFlow data to it:
+        ctx.store_run()
+        cls._log_run(context=ctx, run=run, is_offline=True)
+
+        # Create a rundb in order to update the run's state as completed (can't be done using context)
+        rundb = mlrun.get_run_db()
+        project = run_object.metadata.project
+        uid = run_object.metadata.uid
+        updates = {
+            "status.last_update": now_date().isoformat(),
+            "status.state": "completed",
+        }
+        run_object.status.state = "completed"
+        rundb.update_run(updates, uid, project)
```

Comment:
why? and also you can move to a different function to reduce noise from this function entirely

Reply:
about the why - I see the comment which explains "what" but not "why".

Reply:
we needed to mark the run as completed, this is the way me and my boy @alonmr found to do so

Reply:
what was the issue? why it needed to be completed? .. add few words for your successor who was yet to be born :)

Reply:
ok

Reply:
the issue is that execution and run completion are decoupled. i.e. execution can't determine that the run was completed (like in mpijob where you have multiple executions for a single run)